src/profile-logic/process-profile.js

```diff
@@ -818,7 +818,11 @@ function _adjustMarkerTimestamps(
  * Throws an exception if it encounters an incompatible profile.
  * For a description of the processed format, look at docs/gecko-profile-format.md
  */
-export function processProfile(geckoProfile: GeckoProfile): Profile {
+export function processProfile(rawProfile: Object): Profile {
+  // We may have been given a DevTools profile, in that case extract the Gecko Profile.
+  const geckoProfile: GeckoProfile =
+    'profile' in rawProfile ? rawProfile.profile : rawProfile;
+
```

Comment:
Looking at what's inside the Devtools profile, I wonder if longterm we shouldn't adopt its format instead, as it has some information we don't have at the moment. What do you think ?

But this will work for unblocking people for now.

Reply:
I think the strategy before has been to get the Gecko profile format and the processed format more and more closely aligned. I don't know how much that is happening in reality. I'm personally fine shoving additional information onto the Gecko format, and then taking into account for differences in the processed format. I'd like the processed format to be the canonical target, and then do upgrading as needed.

Although maybe what you are saying is coming from DevTools we don't touch the profile and have something like:

```js
type DevToolsProfile = {
  geckoProfile: GeckoProfile,
  sourceMaps: SourceMaps,
  memoryAllocation: MemoryAllocation,
}
```

Then we'd have:

```js
function processDevToolsProfile(profile: DevToolsProfile): Profile { ... }
```

That seems decently nice, since then platform doesn't have to worry about breaking what DevTools is doing.

I think @mstange would have some opinions here. Although it might be worth breaking out into a separate discussion issue.

Reply:
I don't have a structured opinion, I just think there are valuable information in the Devtools profile we may want to use in the future and that we don't capture with the addon yet.

Reply:
Ideally we'd call this only from "retrieveProfileFromFile"; but well this is some food for thought for later.